sstables/mx/writer.cc

```diff
@@ -1284,7 +1284,21 @@ stop_iteration writer::consume(clustering_row&& cr) {
     ensure_tombstone_is_written();
     ensure_static_row_is_written_if_needed();
     write_clustered(cr);
-    return stop_iteration::no;
+
+    auto stop_at_clustering_boundary = [this] {
+        // will allow size limit to be exceeded for 10%, so we won't perform unnecessary split
+        // of a partition which crossed the limit by a small margin.
+        uint64_t size_threshold_goal = _cfg.max_sstable_size * 1.1;
+        // handle overflow.
+        uint64_t size_threshold = size_threshold_goal < _cfg.max_sstable_size ? std::numeric_limits<uint64_t>::max() : size_threshold_goal;
+        // Check that are enough entries for clustering range metadata to be retrieved on
+        // behalf of the partition being split.
+        bool has_enough_promoted_index_entries = _pi_write_m.promoted_index_size >= 2;
+
+        return get_data_offset() > size_threshold && has_enough_promoted_index_entries;
```

Comment:
can just return `stop_iteration(get_data_offset() > size_threshold && has_enough_promoted_index_entries)`

Reply:
why do we need to define it as a lambda function at all?

Reply:
> why do we need to define it as a lambda function at all?

I decided to put it in a lambda, to avoid creating a private function :-) The logic to decide to split, is standalone.